lib/m2ts/ts-correction-stream.js

```diff
@@ -0,0 +1,88 @@
+/**
+ * mux.js
+ *
+ * Copyright (c) 2016 Brightcove
+ * All rights reserved.
+ *
+ * Accepts program elementary stream (PES) data events and corrects
+ * decode and presentation time stamps to account for a rollover
+ * of the 33 bit value.
+ */
+ 
+'use strict';
+
+var Stream = require('../utils/stream');
+
+var MAX_TS = 8589934592;
+
+var RO_THRESH = 4294967296;
+
+var TSCorrectionStream = function() {
+
+  var
+    reference = {
```

Comment:
Do you think this code could be somewhat simplified if the `TSCorrectionStream` took a parameter on in the constructor to specify the stream type that is handled and ignored packets from every other `type` in the push call?